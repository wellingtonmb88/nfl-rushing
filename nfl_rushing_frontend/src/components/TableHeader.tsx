import React from 'react';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TableSortLabel from '@material-ui/core/TableSortLabel';
import Checkbox from '@material-ui/core/Checkbox';
import { useStyles } from '../styles'

import { Player, HeadCell, EnhancedTableHeadProps } from '../interfaces'

const headCells: HeadCell[] = [
  { id: 'name', numeric: false, disablePadding: true, label: 'Player' },
  { id: 'team', numeric: false, disablePadding: false, label: 'Team' },
  { id: 'position', numeric: false, disablePadding: false, label: 'Pos' },
  { id: 'longestRush', numeric: true, disablePadding: false, label: 'Lng' },
  { id: 'rushingAttempts', numeric: true, disablePadding: false, label: 'Att' },
  { id: 'rushingAttemptsPerGame', numeric: true, disablePadding: false, label: 'Att/G' },
  { id: 'totalRushingYards', numeric: true, disablePadding: false, label: 'Yds' },
  { id: 'rushingAvgYardsPerAttempt', numeric: true, disablePadding: false, label: 'Avg' },
  { id: 'rushingYardsPerGame', numeric: true, disablePadding: false, label: 'Yds/G' },
  { id: 'totalRushingTouchdowns', numeric: true, disablePadding: false, label: 'TD' },
  { id: 'rushingFirstDowns', numeric: true, disablePadding: false, label: '1st' },
  { id: 'rushingFirstDownsPercentage', numeric: true, disablePadding: false, label: '1st%' },
  { id: 'rushingTwentyPlusYeards', numeric: true, disablePadding: false, label: '20+' },
  { id: 'rushingFortyPlusYeards', numeric: true, disablePadding: false, label: '40+' },
  { id: 'rushingFumbles', numeric: true, disablePadding: false, label: 'FUM' },
];

export default function EnhancedTableHead(props: EnhancedTableHeadProps) {
  const { onSelectAllClick, order, orderBy, numSelected, rowCount, rowsPerPage, onRequestSort } = props;
  const classes = useStyles();
  const createSortHandler = (property: keyof Player) => (event: React.MouseEvent<unknown>) => {
    onRequestSort(event, property);
  };

  return (
    <TableHead>
      <TableRow>
        <TableCell padding="checkbox">
          <Checkbox
            indeterminate={numSelected > 0 && numSelected < rowCount}
            checked={rowCount > 0 && (numSelected === rowsPerPage || numSelected === rowCount)}
            onChange={onSelectAllClick}
            inputProps={{ 'aria-label': 'select all players' }}
          />
        </TableCell>
        {headCells.map(headCell => (
          <TableCell
            key={headCell.id}
            align={headCell.numeric ? 'right' : 'left'}
            padding={headCell.disablePadding ? 'none' : 'default'}
            sortDirection={orderBy === headCell.id ? order : false}
          >
            {headCell.id === "longestRush" ||
              headCell.id === "totalRushingYards" ||
              headCell.id === "totalRushingTouchdowns" ?
              (<TableSortLabel
                active={orderBy === headCell.id}
                direction={orderBy === headCell.id ? order : 'asc'}
                onClick={createSortHandler(headCell.id)}
              >
                {headCell.label}
                {orderBy === headCell.id ? (
                  <span className={classes.visuallyHidden}>
                    {order === 'desc' ? 'sorted descending' : 'sorted ascending'}
                  </span>
                ) : null}
              </TableSortLabel>) : headCell.label}

          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  );
}