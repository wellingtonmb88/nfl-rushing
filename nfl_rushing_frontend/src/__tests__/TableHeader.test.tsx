import React from "react";
import { create } from "react-test-renderer";
import TableHeader from '../components/TableHeader'

describe("TableHeader component", () => {
  test("TableHeader with Snapshot", () => {
    const component = create(<TableHeader onSelectAllClick={() => { }} order={'asc'}
      orderBy='' numSelected={2} rowCount={10} rowsPerPage={5} onRequestSort={() => { }} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});