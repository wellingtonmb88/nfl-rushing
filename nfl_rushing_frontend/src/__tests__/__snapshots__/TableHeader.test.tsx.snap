// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader component TableHeader with Snapshot 1`] = `
<thead
  className="MuiTableHead-root"
>
  <tr
    className="MuiTableRow-root MuiTableRow-head"
  >
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-indeterminate MuiIconButton-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            aria-label="select all players"
            checked={false}
            className="PrivateSwitchBase-input-34"
            data-indeterminate={true}
            onChange={[Function]}
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10H7v-2h10v2z"
            />
          </svg>
        </span>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-paddingNone"
      scope="col"
    >
      Player
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
      scope="col"
    >
      Team
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
      scope="col"
    >
      Pos
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Lng
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      Att
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      Att/G
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Yds
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      Avg
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      Yds/G
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        TD
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      1st
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      1st%
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      20+
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      40+
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      FUM
    </th>
  </tr>
</thead>
`;
